src/main/java/com/agilethought/internship/sso/services/ServiceApplicationimpl.java

```diff
@@ -23,24 +28,40 @@
 
 	@Autowired
 	private PasswordEncoder passwordEncoder;
-	
+
 	@Override
 	public UserId createUser(UserDTO userDTO) {
-		if (userDTO.getId() != null)
-			log.info("ServiceApplicationimpl.createUser - id exists");
-			userDTO.setId(null);
-
-		User user = userTransformer.transformer(userDTO);
-		log.info("ServiceApplicationimpl.createUser - users transformed: {}", user);
-		user.setPassword(passwordEncoder.encode(user.getPassword()));
-		String userIdDb = repositoryApplication.save(user).getId();
-		log.info("ServiceApplicationimpl.createUser- User saved  successfully with id: {}", user.getId());
 		UserId userId = new UserId();
```

Comment:
Encapsule all Business Validations and just call one static method called **BusinessValidations.valid().** instead of a lot "if" that makes less readable our code